src/states/elder/mod.rs

```diff
@@ -2048,6 +2055,10 @@ impl Approved for Elder {
         } else if old_pfx.is_extension_of(sec_info.prefix()) {
             self.finalise_prefix_change()?;
             self.send_event(Event::SectionMerged(*sec_info.prefix()), outbox);
+        } else if old_pfx == *sec_info.prefix() {
+            // A pruning happens.
```

Comment:
This does not seem accurate, the prefix will stay the same in most churn events / SectionInfo event since adding/removing a node will generally not lead to a split.

Are the test passing?